api/v1/attributes.proto

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package istio.mixer.v1;
+
+option go_package="istio.io/mixer/api/v1;mixerpb";
+
+// Contains a set of attributes related to an event.
+message Attributes {
```

Comment:
Martin - what do you think about solving the generic handling issue with annotations? E.g.

option (istio.attr) = { name: "source.ip", description: "IPv4 address of the source of the traffic"}
uint32 source_ip = 1;

We're almost certainly going to want some extended metadata for well-known attributes anyway.

I agree this is an early optimization but I strongly suspect well end up having to optimize to eliminate the overhead of the 'name' string field on Attribute pretty quickly. I am fine deferring this until we have some numbers...